mantidproject/mantid
# Kafka run schema update

**Description of work.**

Update to latest schema for run information messages for the Kafka live listener.
Notably, if detector-spectrum mapping is required then the information is sent as part of the "run start" message rather than in a separate message in a separate Kafka topic. Unit tests are updated accordingly.

**To test:**

- Ask me for test files.
- Extract them.
- Run `docker-compose up` to start a Kafka broker and populate it with test data
- Start MantidPlot
- `Load` -> `Live Data` in the "Workspaces" UI
- Set the following options:
  - Instrument: SANS2D
  - Connection: [custom]
  - Listener Type: KafkaEventListener
  - Address String: localhost:9092
  - Starting Time: "Start of Run"
  - Update Every: 5 seconds
  - Processing: "No processing"
  - Preserve events: *tick this*
  - Accumulation Method: Add
  - Post processing: "No processing"
  - At Run Transition: Stop
  - Output Workspace: *enter whatever you like*
- Hit "Run"
- It should all be over fast; the test file is very small. If you look at the instrument view you should see four square detector pixels with a few events.

Fixes #28482. 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
